### DISCUSSION TITLE
shadow dom?

### DISCUSSION THREAD

**Discussion by a-n-d-a-i** [aduros/wasm4#708] [Questions]
Just curious, what is the purpose of using a Shadow DOM?
I ran into some frustration trying to get the image data out of WASM-4's canvas, this appears to be impossible due to being in a Shadow DOM.

I was trying to copy the image data onto a smaller canvas to (further) pixelate my game (for a game jam with very specific technical requirements.) (The alternative was to make my own font (so it could conform to double sized pixels) but I ran out of time.)

I understand that's pretty niche (n=1 ?), but perhaps the magnitude of my frustration makes it worth sharing. (I know nothing about Shadow DOMs except that they ruined my day...)

Apart from that, quite enjoyed working with WASM-4. (It sure beats doing WASM manually like I did for the last jam!)

**Comment by aduros** [comment]
There's no particular reason for Shadow DOM, it's just what Lit is implemented on top of. I agree it can be kind of a pain, but it shouldn't affect most people.
